common/authentication/postgresql/metadata.go

```diff
@@ -112,19 +128,38 @@ func (m *PostgresAuthMetadata) GetPgxPoolConfig() (*pgxpool.Config, error) {
 		// This is because tokens expire, and connections can drop and need to be re-established at any time
 		// Fortunately, we can do this with the "BeforeConnect" hook
 		config.BeforeConnect = func(ctx context.Context, cc *pgx.ConnConfig) error {
-			at, err := tokenCred.GetToken(ctx, policy.TokenRequestOptions{
+			at, errGetAccessToken := tokenCred.GetToken(ctx, policy.TokenRequestOptions{
 				Scopes: []string{
 					m.azureEnv.Cloud.Services[azure.ServiceOSSRDBMS].Audience + "/.default",
 				},
 			})
-			if err != nil {
-				return err
+			if errGetAccessToken != nil {
+				return errGetAccessToken
 			}
 
 			cc.Password = at.Token
 			return nil
 		}
 	}
+	if m.UseAWSIAM {
+		awsRegion, _ := metadata.GetMetadataProperty(m.awsEnv.Metadata, "AWSRegion")
+		if awsRegion == "" {
+			return nil, errors.New("metadata property AWSRegion is missing")
+		}
+		awsAccessKey, _ := metadata.GetMetadataProperty(m.awsEnv.Metadata, "AWSAccessKey")
+		if awsAccessKey == "" {
+			return nil, errors.New("metadata property AWSAccessKey is missing")
+		}
+		awsSecretKey, _ := metadata.GetMetadataProperty(m.awsEnv.Metadata, "AWSSecretKey")
+		if awsSecretKey == "" {
+			return nil, errors.New("metadata property AWSSecretKey is missing")
+		}
 
+		err = awsiam.InitAWSDatabase(context.Background(), config, m.ConnectionString, awsRegion, awsAccessKey, awsSecretKey)
```

Comment:
We need to pass a context so this can be canceled. `context.Background` isn't a valid option. Or @JoshVanL may come chasing after you :)